Wavefront.SDK.CSharp/Common/Application/ApplicationTags.cs

```diff
@@ -101,7 +104,53 @@ public Builder Shard(string shard)
             /// </param>
             public Builder CustomTags(IDictionary<string, string> customTags)
             {
-                this.customTags = customTags;
+                foreach (var tag in customTags)
+                {
+                    this.customTags.Add(tag.Key, tag.Value);
+                }
+                return this;
+            }
+
+            /// <summary>
+            /// Set additional custom tags from environment variables that match the given pattern.
+            /// For instance: APP* etc.
```

Comment:
`^APP.*$` would be better, and perhaps give an example of an env variable it would match to.